File: java/PWMServo/src/main/java/frc/robot/Robot.java
package frc.robot;

import edu.wpi.first.wpilibj.Servo;
import edu.wpi.first.wpilibj.TimedRobot;

/**
 * Moves a PWM servo (port 0) back and forth from the minimum and maximum.
 */
public class Robot extends TimedRobot {

    Servo servo;

  @Override
  public void robotInit() {
      servo = new Servo(0);
  }

  @Override
  public void teleopPeriodic() {

    //Once the servo reaches the min or max throw, set the position target to the opposite side.
    if (servo.get() < 0.1) {
        servo.set(1);
    }
    else if (servo.get() > 0.9) {
        servo.set(0);
    }
  }
}

File: cpp/PWMServo/src/main/cpp/Robot.cpp
#include <frc/TimedRobot.h>
#include <frc/Servo.h>

/**
 * Moves a PWM servo (port 0) back and forth from the minimum and maximum.
 */
class Robot : public frc::TimedRobot
{
public:
  void TeleopPeriodic() override
  {
    //Once the servo reaches the min or max throw, set the position target to the opposite side.
    if (servo.Get() < 0.1)
    {
      servo.Set(1);
    }
    else if (servo.Get() > 0.9)
    {
      servo.Set(0);
    }
  }

private:
  frc::Servo servo{0};
};

#ifndef RUNNING_FRC_TESTS
int main()
{
  return frc::StartRobot<Robot>();
}
#endif
